src/vs/workbench/contrib/terminalContrib/accessibility/browser/terminalAccessibleWidget.ts

```diff
@@ -87,12 +89,21 @@ export abstract class TerminalAccessibleWidget extends DisposableStore {
 		this._element.replaceChildren(this._editorContainer);
 		this._xtermElement.insertAdjacentElement('beforebegin', this._element);
 
-		if (this._focusContextKey) {
-			this._focusTracker = this.add(dom.trackFocus(this._editorContainer));
-			this._focusedContextKey = this._focusContextKey.bindTo(this._contextKeyService);
-			this.add(this._focusTracker.onDidFocus(() => this._focusedContextKey?.set(true)));
-			this.add(this._focusTracker.onDidBlur(() => this._focusedContextKey?.reset()));
-		}
+		this._focusTracker = this.add(dom.trackFocus(this._editorContainer));
+		this._focusedContextKey = rawFocusContextKey.bindTo(this._contextKeyService);
+		this._focusedLastLineContextKey = rawFocusLastLineContextKey.bindTo(this._contextKeyService);
+		this.add(this._focusTracker.onDidFocus(() => {
+			this._focusedContextKey?.set(true);
+			this._focusedLastLineContextKey?.set(this._editorWidget.getSelection()?.positionLineNumber === this._editorWidget.getModel()?.getLineCount());
+		}));
+		this.add(this._focusTracker.onDidBlur(() => {
+			this._focusedContextKey?.reset();
```

Comment:
You might also want to reset this context key when the widget gets disposed.
I'm not sure that onDidBlur gets called when a dom element is removed.

Reply:
https://github.com/microsoft/vscode/pull/191261